### PR TITLE
Fix react warnings

### PR DESCRIPTION
**Pull request by maxyuan6717**
this also improves worksheet performance a bit cuz it cuts down on the calendar rerenders

### CODE REVIEW

File: frontend/src/components/WeekSchedule.js [hunk 1]
```diff
@@ -132,10 +132,12 @@ function WeekSchedule({ showModal, courses, hover_course }) {
...
-  const event_style = {
-    width: '100%',
-    height: '100%',
-  };
+  const event_style = useMemo(() => {
+    return {
+      width: '100%',
+      height: '100%',
+    };
+  }, []);
```
Comment (hsheth2) [inline frontend/src/components/WeekSchedule.js:140]: Move this outside the component function?

File: frontend/src/components/CourseModalOverview.js [hunk 2]
```diff
@@ -106,18 +106,23 @@ const CourseModalOverview = ({ setFilter, filter, setSeason, listing }) => {
     },
   });
   // Hold list of evaluation dictionaries
-  let evaluations = [];
+  let evaluations = useMemo(() => {
+    return [];
+  }, []);
```
Comment (hsheth2) [inline frontend/src/components/CourseModalOverview.js:111]: this seems kinda sus - all the logic to compute the evaluations should be inside here right

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: frontend/src/components/CourseModalOverview.js (hunk 2 above)
Comment (maxyuan6717) [reply frontend/src/components/CourseModalOverview.js]: oh yah im tweaking. just moved `let evaluations = []` into the useMemo for items

**Review by hsheth2 — APPROVED** [review]
LGTM

**Pull request merged**